rotkehlchen/chain/evm/decoding/balancer/utils.py

```diff
@@ -41,17 +42,18 @@
 
 def query_balancer_v1_data(
         inquirer: 'EvmNodeInquirer',
-        cache_type: Literal[CacheType.BALANCER_V1_POOLS],
+        cache_type: Literal[CacheType.BALANCER_V1_POOLS, CacheType.BALANCER_V2_POOLS],
```

Comment:
Pass `version` and `protocol in this function. Also rename it. It's no longer just for v1, right?

This way you will not need to do these if checks you are doing below in the start and in each iteration of the loop